### ISSUE TITLE
HTML coverage report should be generated independent of invoked tasks

### ISSUE THREAD

**Issue by bmuschko** [gogradle/gogradle#202]
The HTML coverage report is generated if you run `./gradlew coverage`. However, I ran into a situation where it is not generated if you first run `./gradlew test` and then `./gradlew coverage`. I also see the following message:

```
:coverage
No coverage profile generated in test task, skip.
```

Any ideas?

**Comment by blindpirate** [comment]
That's weird, is anything abnormal seen in previous test task? Or can I have a reproducible sample?

**Comment by bmuschko** [comment]
I didn't see anything special in the output except the message. I will provide you with a sample soon.

**Comment by amimas** [comment]
I'm seeing some odd behaviour. First, I configured the build to use task alias to avoid task name collision. So, I need to run `gradlew goBuild` instead of `gradlew build`. But, for the coverage task, it seems the alias is set to `goCover`. If I call `gradlew goCoverage`, gradle throws following exception:

```
FAILURE: Build failed with an exception.

* What went wrong:
Task 'goCoverage' not found in root project 'master'. Some candidates are: 'goCover'.
```

Second, even if I run `gradlew goCover`, sometimes I get the error posted by @bmuschko  above. And sometimes I get the following error. Haven't figured out the pattern yet.

```
> Task :my-golang-project:goTest 
Test for my-golang-project finished, 18 completed, 0 failed

flag provided but not defined: -tags
usage: tool [-n] command [args...]

Tool runs the go tool command identified by the arguments.
With no arguments it prints the list of known tools.

The -n flag causes tool to print the command that would be
executed but not execute it.

For more about each tool command, see 'go tool command -h'.

FAILURE: Build failed with an exception.

* What went wrong:
Execution failed for task ':my-golang-project:goCover'.
> Build failed due to return code 2 of: 
  Command:
   /Users/amimas/.gradle/go/binary/1.8.1/go/bin/go tool -tags 'netgo -installsuffix netgo' cover -html=/Users/amimas/gitProjects/docker-plugins/my-golang-project/.gogradle/reports/coverage/profiles/my-golang-project.out -o /Users/amimas/gitProjects/docker-plugins/my-golang-project/.gogradle/reports/coverage/my-golang-project.html
  Env:
   GOEXE=
   GOPATH=/Users/amimas/gitProjects/docker-plugins/my-golang-project/.gogradle/project_gopath
   GOROOT=/Users/amimas/.gradle/go/binary/1.8.1/go
   GOOS=darwin
   GOARCH=amd64
```

From the first post by @bmuschko it sounds like the coverage report should be generated if I only run `gradlew goCover` (without any other tasks), but I'm not able to get any coverage report generated.

**Comment by amimas** [comment]
Actually, I was just doing some trial and error and it seems the error I reported above is due to build tags in my `build.gradle`

```gradle
golang {
	packagePath = project.name
	goVersion = '1.8.1'
	goExecutable = 'IGNORE_LOCAL'
	// buildTags = ['netgo -installsuffix netgo']
}
```
If I don't include the `buildTags`, then running `gradlew goClean goCover` seems to produce the coverage report for me. And if I include the build tags, then I get the error I reported. But, not really sure why the subsequent `gradlew goCover` command reports the other issue (No Coverage profile generated) instead of the same error.

Also noticed, once the coverage report is generated, if I delete the `.gogradle/reports/coverage/` directory and then run `gradlew goCover`, it gives the same message "No Coverage profile generated". Am I mistaken to expect the coverage report in this scenario?

I will try your suggestion mentioned above.

**Comment by blindpirate** [comment]
Yes, I guess your issue is due to this PR: https://github.com/gogradle/gogradle/pull/174

> No coverage profile generated in test task, skip.

This is definitely a bug. When you run `cover` task which depends on `test` task, `test` task will run first but `.gogradle/reports/coverage` isn't marked as its output dir so `test` task will be treated as `UP-TO-DATE` and skipped. A workaround is always run `goTest goCover --rerun-tasks`.

Thanks for reporting. I think this is worth a `0.8.2` release.

**Comment by amimas** [comment]
@blindpirate - I ran the command you suggested with the env variables set. I get this from the command:

```
flag provided but not defined: -tags
Usage of 'go tool cover':
Given a coverage profile produced by 'go test':
	go test -coverprofile=c.out

Open a web browser displaying annotated source code:
	go tool cover -html=c.out

Write out an HTML file instead of launching a web browser:
	go tool cover -html=c.out -o coverage.html

Display coverage percentages to stdout for each function:
	go tool cover -func=c.out

Finally, to generate modified source code with coverage annotations
(what go test -cover does):
	go tool cover -mode=set -var=CoverageVariableName program.go

Flags:
  -func string
    	output coverage profile information for each function
  -html string
    	generate HTML representation of coverage profile
  -mode string
    	coverage mode: set, count, atomic
  -o string
    	file for output; default: stdout
  -var string
    	name of coverage variable to generate (default "GoCover")

  Only one of -html, -func, or -mode may be set.
```
Does the `cover` task currently not work with buildTags? BTW, will you rename the task alias name to `goCoverage`? I think that will make it consistent with others.

And if you are thinking of another release due to this bug, may I ask if it would be possible for the task to report to STDOUT about the total coverage percentage? I was thinking of using it in my Gitlab CI, which can read stdout of a build and show coverage report to a project's page.

**Comment by blindpirate** [comment]
Seems `go cover` doesn't support buildTags, I need to do some investigation. 

I think `cover` and `goCover` would be better because the tool is named `go tool cover`.

As for outputting coverage data to stdout, this looks like a feature. Would you mind opening another issue to describe your demand in detail? Thanks.

**Comment by amimas** [comment]
I see your point about cover and goCover tadk names. I think that makes sense. It should be ok as long as there's consistency. Hopefully it won't have too much impact on others.

I'll open a separate issue regarding the other feature request.

**Comment by blindpirate** [comment]
Closed by `0.9`. Thanks for reporting.
